packages/hoppscotch-common/src/helpers/preRequest.ts

```diff
@@ -8,11 +8,63 @@ import {
   getGlobalVariables,
 } from "~/newstore/environments"
 import { TestResult } from "@hoppscotch/js-sandbox"
+import { getService } from "~/modules/dioc"
+import { SecretEnvironmentService } from "~/services/secret-environment.service"
 
-export const getCombinedEnvVariables = () => ({
-  global: cloneDeep(getGlobalVariables()),
-  selected: cloneDeep(getCurrentEnvironment().variables),
-})
+const secretEnvironmentService = getService(SecretEnvironmentService)
+
+const unsecretEnvironments = (
+  global: Environment["variables"],
+  selected: Environment
+) => {
+  const resolvedGlobalWithSecrets = global.map((globalVar, index) => {
+    const secretVar = secretEnvironmentService.getSecretEnvironmentVariable(
+      "Global",
+      index
+    )
+    if (secretVar) {
+      return {
+        ...globalVar,
+        secret: false,
+        value: secretVar.value,
+      }
+    }
+    return globalVar
```

Comment:
```suggestion
    return {
      ...globalVar,
      value:
        (
          globalVar as {
            value: string
            key: string
            secret: false
          }
        ).value ?? "",
    }
```

I noticed once while importing an environment with secret values being empty, and attempting to update it from the scripting context didn't work as expected since the `value` field wasn't present in the incoming data. Would it be helpful to ensure the `value` field exists for the fallback?